tests/Unit/Core/Parser/TemplateParserPatternTest.php

```diff
@@ -461,6 +461,112 @@ public function SPLIT_PATTERN_SHORTHANDSYNTAX_ARRAY_PARTS_matchesQuotedKeys()
     /**
      * @return array
      */
+    public function dataProviderValidArrayExpressionsBeginningAndEndingOnDigits()
+    {
+        return [
+            [
+                'expression' => '{baz: foo.bar1}',
+                'expected' => [
+                    0 => [
+                        0 => 'baz: foo.bar1',
+                        'ArrayPart' => 'baz: foo.bar1',
+                        1 => 'baz: foo.bar1',
+                        'Key' => 'baz',
+                        2 => 'baz',
+                        'QuotedString' => '',
+                        3 => '',
+                        'VariableIdentifier' => 'foo.bar1',
+                        4 => 'foo.bar1'
+                    ]
+                ]
+            ],
+            [
+                'expression' => '{"foo2": "bar2"}',
+                'expected' => [
+                    0 => [
+                        0 => '"foo2": "bar2"',
+                        'ArrayPart' => '"foo2": "bar2"',
+                        1 => '"foo2": "bar2"',
+                        'Key' => '"foo2"',
+                        2 => '"foo2"',
+                        'QuotedString' => '"bar2"',
+                        3 => '"bar2"',
+                    ]
+                ]
+            ],
+            [
+                'expression' => '{foo3: bar.baz3}',
+                'expected' => [
+                    0 => [
+                        0 => 'foo3: bar.baz3',
+                        'ArrayPart' => 'foo3: bar.baz3',
+                        1 => 'foo3: bar.baz3',
+                        'Key' => 'foo3',
+                        2 => 'foo3',
+                        'QuotedString' => '',
+                        3 => '',
+                        'VariableIdentifier' => 'bar.baz3',
+                        4 => 'bar.baz3'
+                    ]
+                ]
+            ],
+            [
+                'expression' => '{\'foo4\': 5}',
```

Comment:
I'm not 100% sure about this one... we do support *double*quoted strings (so we match JSON there) but single-quoted strings I think should only be possible as values. Hopefully it isn't some coincidence that lets this work.

No objection if a FEATURE does this later on, pending what @liayn is looking into. But it does not seem to be part of this bugfix?